benchmark/simulate_hgts.py

```diff
@@ -0,0 +1,627 @@
+                # insert gene (protein)
+                hgt_gene = "%s_hgt_n" % gene_donor_label
+                genes_recip[hgt_gene] =\
+                    [genes_donor[gene_donor_label][0], idx_recip, idx_end,
+                     genes_donor[gene_donor_label][3]]
+                # insert gene (nucleotide)
+                seq_recip = Sequence(
+                    str(seq_recip[:idx_recip]) +
+                    str(seq_donor[genes_donor[gene_donor_label][1]:
+                        genes_donor[gene_donor_label][2]]) +
+                    str(seq_recip[idx_recip:]))
+                # write HGTs to log file
+                log_f.write(
+                    "n\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n" %
+                    (gene_donor_label,
+                     genes_donor[gene_donor_label][1],
+                     genes_donor[gene_donor_label][2],
+                     hgt_gene,
+                     idx_recip,
+                     idx_end,
+                     genes_donor[gene_donor_label][3]))
+                break
+            # try next open region
+            idx_recip = gene_positions_s[y+1][1] + 1
+
+    return seq_recip
+
+
+def write_results(genes_donor,
+                  donor_genome_fp,
+                  genes_recip,
+                  recip_genome_fp,
+                  seq_donor,
+                  seq_recip,
+                  output_dir):
+    """Write donor and recipient genomes to FASTA files (nucl and protein).
+
+    Parameters
+    ----------
+    genes_donor: dictionary
+    seq_donor: skbio.sequence.Sequence
+        Sequence object for donor genome
+    genes_recip: dictionary
+    seq_recip: skbio.sequence.Sequence
+        Sequence object for recipient genome
+    """
+    # output dir for simulated results
+    simulated_dir = join(output_dir, "simulated")
+    if not exists(simulated_dir):
+        makedirs(simulated_dir)
+    donor_genome_aa_fp = join(
+        simulated_dir, "%s.faa" % basename(splitext(donor_genome_fp)[0]))
+    recip_genome_aa_fp = join(
+        simulated_dir, "%s.faa" % basename(splitext(recip_genome_fp)[0]))
+    with open(donor_genome_aa_fp, 'w') as donor_genome_aa_f:
+        for gene in genes_donor:
+            donor_genome_aa_f.write(
+                ">%s\n%s\n" % (gene, genes_donor[gene][0]))
+    with open(recip_genome_aa_fp, 'w') as recip_genome_aa_f:
+        for gene in genes_recip:
+            recip_genome_aa_f.write(
+                ">%s\n%s\n" % (gene, genes_recip[gene][0]))
+    donor_genome_nucl_fp = join(
+        simulated_dir, "%s.fna" % basename(splitext(donor_genome_fp)[0]))
+    recip_genome_nucl_fp = join(
+        simulated_dir, "%s.fna" % basename(splitext(recip_genome_fp)[0]))
+    seq_donor.write(donor_genome_nucl_fp, format='fasta')
+    seq_recip.write(recip_genome_nucl_fp, format='fasta')
+
+    return (donor_genome_nucl_fp, donor_genome_aa_fp, recip_genome_nucl_fp,
+            recip_genome_aa_fp)
+
+
+def simulate_hgts(seq_donor,
```

Comment:
Hmm, I'm using 6/12 parameters (output_dir, donor_genome_fp, recip_genome_fp, genes_donor, genes_recip, orthologous_rep_prob) passed to ```foo``` in ```foo```. I'll need to unpack them with **kwargs, does it still make sense to use it?